plutus-chain-index/src/Plutus/ChainIndex/SyncStats.hs

```diff
@@ -0,0 +1,141 @@
+    { syncStateSyncLog :: SyncState -- ^ State of the syncing
+    , syncStatsSyncLog :: SyncStats -- ^ Stats of the syncing
+    , delaySyncLog     :: Second -- ^ Delay in seconds used to accumulate log events
+    }
+    deriving stock (Eq, Show, Generic)
+    deriving anyclass (FromJSON, ToJSON, ToObject)
+
+instance Pretty SyncLog where
+  pretty = \case
+    SyncLog syncState
+            (SyncStats numRollForward numRollBackwards chainSyncPoint _)
+            seconds ->
+        let currentTipMsg NotSyncing = ""
+            currentTipMsg _          = "Current tip is" <+> pretty chainSyncPoint
+         in
+            pretty syncState
+                <+> "Applied"
+                <+> pretty numRollForward
+                <+> "blocks"
+                <> comma
+                <+> pretty numRollBackwards
+                <+> "rollbacks in the last"
+                <+> viaShow seconds
+                <> "."
+                <+> currentTipMsg syncState
+
+data SyncState = Synced | Syncing Double | NotSyncing
+    deriving stock (Eq, Show, Generic)
+    deriving anyclass (FromJSON, ToJSON, ToObject)
+
+instance Pretty SyncState where
+  pretty = \case
+    Synced      -> "Still in sync."
+    Syncing pct -> "Syncing (" <> pretty (printf "%.2f" pct :: String) <> "%)."
+    NotSyncing  -> "Not syncing."
+
+-- | Read 'ChainSyncEvent's for the 'TChan' every 30 seconds and log syncing summary.
+logProgress :: forall effs.
+    ( Member (LogMsg SyncLog) effs
+    , LastMember IO effs
+    )
+    => TChan SyncStats
+    -> Eff effs ()
+logProgress broadcastChan = do
+    chan <- liftIO $ atomically $ dupTChan broadcastChan
+    go chan 30_000_000 -- 30s
+  where
+    go chan delay = do
+        liftIO $ threadDelay (fromIntegral delay)
+        syncStats <- liftIO $ foldTChanUntilEmpty chan
+        let syncState = getSyncStateFromStats syncStats
+        case syncState of
+          NotSyncing -> do
+              logWarn $ SyncLog syncState syncStats (fromMicroseconds delay)
+              go chan 300_000_000 -- 300s
+          Synced -> do
+              logInfo $ SyncLog syncState syncStats (fromMicroseconds delay)
+              go chan 300_000_000 -- 300s
+          Syncing _ -> do
+              logInfo $ SyncLog syncState syncStats (fromMicroseconds delay)
+              go chan 30_000_000 -- 30s
+
+-- | Get the 'SyncState' for a 'SyncState'.
+--
+-- TODO: The syncing percentage is valid when the node is already fully synced.
+-- But when the node and chain-index are started at the same time, the syncing
+-- percentage is not a valid number considering the actual tip of the node.
+-- Find a better way to calculate this percentage.
+getSyncStateFromStats :: SyncStats -> SyncState
+getSyncStateFromStats (SyncStats _ _ chainSyncPoint nodePoint) =
+    case (chainSyncPoint, nodePoint) of
+        (_, PointAtGenesis) -> NotSyncing
+        (CI.PointAtGenesis, CI.Point _ _) -> NotSyncing
+        (p1, p2) | p1 == p2 -> Synced
```

Comment:
Since new transactions keep coming in, I wonder if this state is ever reached. In my old code this was fixed implicitly because rounding would show `100%` at some point and this would then never be shown again. I believe the cardano wallet has defined a margin of error somewhere that says: if we're this close to the node point, we call syncing done.

Reply:
Mmm.. Then I can do something similar: `(p1, p2) | p2 - p1 < delta -> Synced`

Reply:
@sjoerdvisscher I don't know the best number, so I did:

```
        (CI.Point (Slot chainSyncSlot) _, CI.Point (Slot nodeSlot) _) | nodeSlot - chainSyncSlot < 100 -> Synced
```

Should be reasonable for now.